packages/hooks/use-global-config/index.ts

```diff
@@ -53,11 +52,22 @@ export const provideGlobalConfig = (
   const context = computed(() => {
     const cfg = unref(config)
     if (!oldConfig?.value) return cfg
-    return merge(oldConfig.value, cfg)
+    return mergeConfig(oldConfig.value, cfg)
   })
   provideFn(configProviderContextKey, context)
   if (global || !globalConfig.value) {
     globalConfig.value = context.value
   }
   return context
 }
+
+const mergeConfig = <T extends Record<string, any>>(a: T, b: T) => {
```

Comment:
**Question:** since this method is a dedicated method only targeted for merging the config, should we skip the generic type definition part and assigning the parameter the type of the config?